src/sdk/namespace_fs.js

```diff
@@ -915,6 +915,17 @@ class NamespaceFS {
                 }
             };
 
+            const format_key_name = obj_info => {
+                if (this._is_hidden_version_path(obj_info.key)) {
+                    obj_info.key = path.normalize(obj_info.key.replace(HIDDEN_VERSIONS_PATH + '/', ''));
+                    obj_info.key = _get_filename(obj_info.key);
+                    set_latest_delete_marker(obj_info);
```

Comment:
should set_latest_delete_marker() function be a part of a function called format_key_name() ?

Reply:
right, it shouldn't. also putting it inside _is_hidden_version_path is not important. I moved it outside